### ISSUE TITLE
[2.0.0-rc.15] gatsby-ssr.js: pass props to bodyComponent in replaceRenderer

### ISSUE THREAD

**Issue by HriBB** [gatsbyjs/gatsby#8018]
I am unable to pass extra props to `bodyComponent` inside `replaceRenderrer` with `gatsby@2.0.0-rc.15`.

```jsx
const React = require('react')
const { renderToString } = require('react-dom/server')
const { JssProvider } = require('react-jss')

const postcss = require('postcss')
const autoprefixer = require('autoprefixer')
const csswring = require('csswring')

const { AdyenProvider } = require('providers/adyen')
const { CartProvider } = require('providers/cart')
const { DrawerProvider } = require('providers/drawer')

const getPageContext = require('utils/getPageContext').default
const Layout = require('components/layout/Layout').default

const prefixer = postcss([autoprefixer])
const minifier = postcss([csswring])

exports.wrapRootElement = ({ element }) => {
  return (
    <AdyenProvider>
      <CartProvider>
        <DrawerProvider>
          {element}
        </DrawerProvider>
      </CartProvider>
    </AdyenProvider>
  )
}

exports.wrapPageElement = ({ element, props }) => {
  console.log(`\nmuiPageContext`, props.muiPageContext) 
  return (
    <Layout {...props}>{element}</Layout>
  )
}

exports.replaceRenderer = ({ bodyComponent, replaceBodyHTMLString, setHeadComponents }) => {
  const muiPageContext = getPageContext()

  const bodyHTML = renderToString(
    <JssProvider
      registry={muiPageContext.sheetsRegistry}
      generateClassName={muiPageContext.generateClassName}
    >
      {React.cloneElement(bodyComponent, {
        muiPageContext,
      })}
    </JssProvider>,
  )

  replaceBodyHTMLString(bodyHTML)

  // extract and minify styles
  const css = muiPageContext.sheetsRegistry.toString()
  const options = { from: undefined }
  const prefixed = prefixer.process(css, options)
  const minified = minifier.process(prefixed.css, options)

  setHeadComponents([
    <style
      type={'text/css'}
      id={'server-side-jss'}
      key={'server-side-jss'}
      dangerouslySetInnerHTML={{ __html: minified.css }}
    />,
  ])
}
```

`props` inside `wrapPageElement` do not contain `muiPageContext` when doing SSR.

**Comment by kakadiadarpan** [comment]
Hi @HriBB, as per the code above I understand that you are trying to pass props to `bodyComponent`. You should use [`setBodyProps`](https://next.gatsbyjs.org/docs/ssr-apis/#replaceRenderer) instead to pass props to the `bodyComponent`

**Comment by HriBB** [comment]
So I took another stab at this ... @kakadiadarpan `setBodyProps` is for `Html` component I believe. https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/static-entry.js#L370

I am trying to pass props down to my page component from the `replaceRenderer`, specifically `@material-ui` context required for SSR. 

It used to work in `gatsby@2.0.0-beta.59`:
```jsx
exports.replaceRenderer = ({ bodyComponent, replaceBodyHTMLString, setHeadComponents }) => {
  const muiPageContext = getPageContext()

  const bodyHTML = renderToString(
    <JssProvider
      registry={muiPageContext.sheetsRegistry}
      generateClassName={muiPageContext.generateClassName}
    >
      {React.cloneElement(bodyComponent, {
        muiPageContext, // <<<<<<< HERE
      })}
    </JssProvider>,
  )

  replaceBodyHTMLString(bodyHTML)

  // extract and minify styles
  const css = muiPageContext.sheetsRegistry.toString()
  const options = { from: undefined }
  const prefixed = prefixer.process(css, options)
  const minified = minifier.process(prefixed.css, options)

  setHeadComponents([
    <style
      type={'text/css'}
      id={'server-side-jss'}
      key={'server-side-jss'}
      dangerouslySetInnerHTML={{ __html: minified.css }}
    />,
  ])
}
```

But then @pieh implemented the new `wrapRootElement` and `wrapElement` and now it does not work anymore. I'm not sure if this is a bug or not.

**Comment by HriBB** [comment]
I upgraded `gatsby` to version `2.0.0-rc.15`, @material-ui/core` to version `3.0.3` and updated my code following the [official mui gatsby example](https://github.com/mui-org/material-ui/tree/master/examples/gatsby) and now it works.

The example uses `JssProvider` in `replaceRenderer` and `withRoot`. If I understand things correctly, two `JssProviders` are used to render on the server, which is not perfect, but it works.

Closing the issue.
